go/pserver/etcd_cclient.go

```diff
@@ -0,0 +1,128 @@
+package pserver
+
+import (
+	"context"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	DefaultEtcdTimeout time.Duration = 5 * time.Second
+)
+
+type EtcdCClient interface {
+	Desired() int
+	List() []Server
+}
+
+// TODO(Longfei)
+// 1. add watcher to watch the change state of pservers)
+// 1. add etcd lock)
+type EtcdCClientImpl struct {
```

Comment:
It's a very common pattern in Java: create interface, and write the implementation, this is due to the explicit interface in Java. It's not required for Go: Go has implicit interface. Please see: https://medium.com/@cep21/preemptive-interface-anti-pattern-in-go-54c18ac0668a Unnecessary interface only adds complication to the program.

I think we need to stick to "accept interface, return struct" unless there is good reason not to.
In this case we need to remove `type EtcdCClient interface`, rename `EtcdCClientImpl` to `EtcdCClient`, and make `func NewEtcdCClient(endpoints string) (*EtcdCClient, error)` return pointer to the struct.